### DISCUSSION TITLE
Passing arguments to pyinstaller when using `gen --pack` cmd

### DISCUSSION THREAD

**Discussion by Eboubaker** [dashingsoft/pyarmor#1828] [Q&A] [answered]
In pyamror-7 you can do this: https://github.com/dashingsoft/pyarmor/issues/71

How can i pass the `--noconfirm` option to pyinstaller when using `pyarmror gen --pack main.spec -r main.py` which is version 8

**Comment by jondy** [comment, marked as answer]
```
$ pyarmor cfg pack:pyi_options = " --noconfirm"
```
More information about pack
https://pyarmor.readthedocs.io/en/latest/topic/repack.html

**Reply by Eboubaker** [reply]
Thank you I am using it inside of a python script this also worked based on your input:

```python
import pyarmor.cli.__main__
pyarmor.cli.__main__.main_entry([
    'cfg',
    'pack:pyi_options="--noconfirm"',
])
pyarmor.cli.__main__.main_entry([
    'gen',
    '--pack',
    name(oscript) + '.exe.spec',
    '-r',
    script
])
```
it now does a cleaning of the dist folder before building
